Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -765,6 +774,28 @@ fileprivate extension Compiler.ByteCodeGen {
     case .characterPredicate:
       throw Unsupported("character predicates")
 
+    case .typeErase(let child):
+      // FIXME: This is a workaround for `Regex<AnyRegexOutput>` not working in
+      // the DSL. This separates any `Regex<AnyRegexOutput>` into its own
+      // compilation unit, but is less efficient. We should instead represent
+      // output type erasure in the matching engine (`beginTypeErase`,
+      // `endTypeErase`).
+      //
+      // Long-term design:
+      //   beginTypeErase
+      //   <code for child>
+      //   endTypeErase
+      let program = try Compiler(tree: DSLTree(child)).emit()
+      let executor = Executor(program: program)
+      return emitMatcher { input, startIndex, range in
```

Comment:
Matchers don't get the subject bounds, so I don't see how this would support anchors that refer to the subject bounds. We _could_ have a different matcher interface or else add subject bounds as an additional parameter for the internal code (probably not API though).